lib/emqx_elixir_plugin/elixir_plugin_body.ex

```diff
@@ -16,25 +16,23 @@
 
 
 defmodule EmqxElixirPlugin.Body do
-    
-    require Record    
 
-    Record.defrecord(:message, Record.extract(:message, from_lib: "emqx/include/emqx.hrl"))
+    require Record
 
     def hook_add(a, b, c) do
         :emqx_hooks.add(a, b, c)
     end
-    
+
     def hook_del(a, b) do
         :emqx_hooks.del(a, b)
     end
 
     def load(env) do
         # uncomment the hooks that you want, and implement its callback
-        
+
         #hook_add(:"client.authenticate",  &EmqxElixirPlugin.Body.on_client_authenticate/2, [env])
         #hook_add(:"client.check_acl",     &EmqxElixirPlugin.Body.on_client_check_acl/5,    [env])
-        #hook_add(:"message.publish",      &EmqxElixirPlugin.Body.on_message_publish/2,     [env])
+        hook_add(:"message.publish",      &EmqxElixirPlugin.Body.on_message_publish/2,     [env])
         #hook_add(:"message.deliver",      &EmqxElixirPlugin.Body.on_message_deliver/3,     [env])
         #hook_add(:"message.acked",        &EmqxElixirPlugin.Body.on_message_acked/3,       [env])
         #hook_add(:"client.connected",     &EmqxElixirPlugin.Body.on_client_connected/4,    [env])
```

Comment:
on_client_connected in now arity 3, not 4.
https://github.com/emqx/emqx-plugin-template/blob/8d0441fa55c66c3bad3b77fecf16a665a3ba1962/src/emqx_plugin_template.erl#L89

Reply:
dropped all unnecessary hooks for demo